public/app/features/explore/hooks/useStateSync/synchronizer/init.ts

```diff
@@ -0,0 +1,126 @@
+import { identity, isEqual } from 'lodash';
+import { MutableRefObject } from 'react';
+
+import { EventBusSrv } from '@grafana/data';
+import { LocationService } from '@grafana/runtime';
+import { DataQuery } from '@grafana/schema';
+import { initializeExplore } from 'app/features/explore/state/explorePane';
+import { clearPanes, syncTimesAction } from 'app/features/explore/state/main';
+import { fromURLRange } from 'app/features/explore/state/utils';
+import { withUniqueRefIds } from 'app/features/explore/utils/queries';
+import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
+import { ThunkDispatch } from 'app/types';
+
+import { getUrlStateFromPaneState } from '../index';
+import {
+  getDefaultQuery,
+  getPaneDatasource,
+  getQueryFilter,
+  isMixedDatasource,
+  removeQueriesWithInvalidDatasource,
+} from '../internal.utils';
+import { ExploreURLV1 } from '../migrators/v1';
+
+export function initializeFromURL(
+  urlState: ExploreURLV1,
+  initState: MutableRefObject<'pending' | 'done' | 'notstarted'>,
```

Comment:
given we are now breaking this into multiple files, what about exporting a type somewhere like

```ts
type InitState = 'notstarted' | 'pending' | 'done';
```

to use here and in `public/app/features/explore/hooks/useStateSync/synchronizer/toURL.ts` as `MutableRefObject<InitState>` ?

Reply:
Good point! Updated in 1586cc5e02e2841cd26cbdd0e9134ba17360959f.